daphne/http_protocol.py

```diff
@@ -185,15 +185,31 @@ def process(self):
                     # Not much we can do, the request is prematurely abandoned.
                     return
                 # Run application against request
-                self.application_queue.put_nowait(
-                    {"type": "http.request", "body": self.content.read()}
-                )
+                buffer_size = self.server.request_buffer_size
+                self.consume_body(buffer_size)
+
         except Exception:
             logger.error(traceback.format_exc())
             self.basic_error(
                 500, b"Internal Server Error", "Daphne HTTP processing error"
             )
 
+    def consume_body(self, buffer_size):
+        """
+        Consumes body in chunks.
+        """
+
+        chunk = self.content.read(buffer_size)
+        if not chunk:
+            self.application_queue.put_nowait({"type": "http.request", "body": chunk})
+            return
+        incomplete_chunk = len(chunk) < buffer_size
+        payload = {"type": "http.request", "body": chunk, "more_body": False}
+        if not incomplete_chunk:
+            payload["more_body"] = True
```

Comment:
Can we combine the `not chunk` and `incomplete_chunk` parts? 

I didn't test, but something like: 

```python
while True: 
    chunk = self.content.read(buffer_size)       
    more_body = not (len(chunk) < buffer_size)
    payload = {"type": "http.request", "body": chunk, "more_body": more_body}
    self.application_queue.put_nowait(payload)
    if not more_body:
        break
```

I'd then be inclined to inline that at line189

Reply:
Ah yeah, I don't like the code so much. This is much better. The only "downside" would be that more_body will still be in the message when there's nothing more to consume, which is not the cleanest way to do things. But I reckon this is still valid per ASGI's spec so that's fine by me.

Reply:
Done